readthedocs/doc_builder/python_environments.py

```diff
@@ -168,7 +163,7 @@ def save_environment_json(self):
         with open(self.environment_json_path(), 'w') as fpath:
             # Compatibility for Py2 and Py3. ``io.TextIOWrapper`` expects
             # unicode but ``json.dumps`` returns str in Py2.
-            fpath.write(unicode(json.dumps(data)))
+            fpath.write(str(json.dumps(data)))
```

Comment:
@humitos explains that this might not be the fix we want:

https://github.com/rtfd/readthedocs.org/issues/3516#issuecomment-357849119

Do we need to write via a handle instantiated through `codecs` perhaps?

Reply:
What about six.text_type ?